client/state/selectors/get-activity-logs.js

```diff
@@ -1,19 +1,34 @@
+/** @format *
 /**
  * External dependencies
  */
 import { get } from 'lodash';
 
 /**
+ * Internal dependencies
+ */
+import ActivityQueryManager from 'lib/query-manager/activity';
+import createSelector from 'lib/create-selector';
+
+/**
  * Returns list of Activity Log items.
  *
- * @param {Object} state Global state tree
- * @param {number|string} siteId the site ID
- * @return {?array} Activity log item objects. Null if no data.
+ * @param  {Object}        state  Global state tree
+ * @param  {number|string} siteId the site ID
+ * @param  {?Object}       query  Optional. Query object, passed to ActivityQueryManager.
+ * @return {?array}               Activity log item objects. Null if no data.
  */
-export default function getActivityLogs( state, siteId ) {
-	return get( state, [
-		'activityLog',
-		'logItems',
-		siteId,
-	], null );
-}
+const getActivityLogs = createSelector(
+	( state, siteId, query ) => {
+		const manager = get( state, [ 'activityLog', 'logItems', siteId ], null );
+		if ( ! manager ) {
+			return null;
+		}
+
+		return manager.getItems( query );
+	},
+	( state, siteId ) => get( state, [ 'activityLog', 'logItems', siteId ] ),
+	( state, siteId, query ) => ActivityQueryManager.QueryKey.stringify( query )
```

Comment:
Should the returned cache key include the `siteId`?

Reply:
ok, it's part of the `query`. Got it 👍

Reply:
Although, `query` is optional, is that correct? So how does the cache key look if no `query` is supplied?

Reply:
I spoke off GH with @seear a bit and we decided to simple remove the memoization. There appears to be very limited benefit.